resolwe/flow/executors/__init__.py

```diff
@@ -315,7 +330,7 @@ def run(self, data_id, script, verbosity=1):
                             for entity in entities:
                                 entity.collections.add(collection)
 
-            if process_rc == 0:
+            if process_rc == 0 and not self.failed:
```

Comment:
Couldn't you simply check if the data does not have status ERROR instead of `self.failed`?